backend/api/views/main.py

```diff
@@ -1,17 +1,22 @@
 from flask import Blueprint
-from flask import request
+from flask import request, send_from_directory, current_app
+
 from api.core import create_response, serialize_list, logger
 from api.models import Question, db
+import os
 
 main = Blueprint("main", __name__)
 
-
-# function that is called when you visit /
-@main.route("/")
-def index():
-    # access the logger with the logger from api.core and uses the standard logging module
-    logger.info("Hello World!")
-    return "<h1>Hello World!</h1>"
+# serve the react frontend
+@main.route("/", defaults={"path": ""})
+@main.route("/<path:path>")
+def serve_frontend(path):
+    static_folder = current_app.static_folder
+    print("using this new path now", static_folder)
+    if path != "" and os.path.exists(static_folder + "/" + path):
+        return send_from_directory(static_folder, path)
+    else:
+        return send_from_directory(static_folder, "index.html")
```

Comment:
Nice file server :+1:

Reply:
Just a thought: could they still manage to escape from static folder location and read arbitrary files from the server? Even though you concat static_folder onto the given path, what if they then put "../" in the name? Idk I could be wrong

Reply:
oh no they can't because flask's send_from_directory restricts them
I believe if you use something like send_static_file or send_file then you run into this security issue where they could theoretically look at backend code or anything they want really